File: py/src/practise150/p12_three_sum_test.py
from src.practise150.p12_three_sum import Solution

def test_three_sum():
    test_cases = [
        ([-1,0,1,2,-1,-4], [[-1,-1,2],[-1,0,1]]),
        ([0,1,1], []),
        ([0,0,0], [[0,0,0]])
    ]
    
    s = Solution()
    for test_case in test_cases:
        result = s.three_sum(test_case[0])
        assert result == test_case[1]

File: go/practise150/p12_three_sum_test.go
package practise150

import (
	"fmt"
	"reflect"
	"testing"
)

func TestThreeSum(t *testing.T) {
	test_cases := []struct {
		input    []int
		expected [][]int
	}{
		{
			input:    []int{-1, 0, 1, 2, -1, -4},
			expected: [][]int{{-1, -1, 2}, {-1, 0, 1}},
		},
		{
			input:    []int{0, 1, 1},
			expected: [][]int{},
		},
		{
			input:    []int{0, 0, 0},
			expected: [][]int{{0, 0, 0}},
		},
	}

	for idx, test_case := range test_cases {
		t.Run(fmt.Sprintf("Test_case_no:%d", idx+1), func(t *testing.T) {
			actual := ThreeSum(test_case.input)
			if !reflect.DeepEqual(actual, test_case.expected) {
				t.Errorf("Testing Three sum function. Input: %v, Expected: %v, Actual: %v.", test_case.input, test_case.expected, actual)
			}
		})
	}
}
